docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+>>> api = HfApi()
+
+# List all models.
+>>> api.list_models()
+
+# List only text classification models.
+>>> api.list_models(filter="text-classification")
+
+# List only Russian models compatible with PyTorch.
+>>> api.list_models(filter=("ru", "pytorch"))
+
+# List only the models trained on the "common_voice" dataset.
+>>> api.list_models(filter="dataset:common_voice")
+
+# List only the models from the spaCy library.
+>>> api.list_models(filter="spacy")
+```
+
+Explore available public datasets with `list_datasets`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List only text classification datasets.
+>>> api.list_datasets(filter="task_categories:text-classification")
+
+# List only datasets in Russian for language modeling.
+>>> api.list_datasets(filter=("languages:ru", "task_ids:language-modeling"))
+```
+
+### Inspect model or dataset metadata
+
+Get important information about a model or dataset as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# Get metadata of a single model.
+>>> api.model_info("distilbert-base-uncased")
+
+# Get metadata of a single dataset.
+>>> api.dataset_info("glue")
+```
+
+### Create a repository
+
+Create a repository with `create_repo` and give it a name with the `name` parameter:
+
+1. Get your Hugging Face API token:
+
+   ```python
+   >>> from huggingface_hub import HfFolder
+   >>> token = HfFolder().get_token()
+   ```
+2. Use `create_repo` to create your repository:
+
+   ```python
+   >>> from huggingface_hub import HfApi
+   >>> api = HfApi()
+   >>> api.create_repo(token=token, name="test-model")
+   'https://huggingface.co/lysandre/test-model'
+   ```
+
+### Delete a repository
+
+Delete a repository with `delete_repo`. Make sure you are certain you want to delete a repository because this is an irreversible process!
+
+Pass your token and the name of the repository to `delete_repo`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.delete_repo(token=token, name=REPO_NAME)
+```
+
+Delete a dataset repository by adding the `repo_type` parameter:
+
+```python
+>>> api.delete_repo(token=token, name=REPO_NAME, repo_type="dataset")
+```
+
+### Change repository visibility
+
+A repository can be public or private. A private repository is only visible to you or members of your organization. Change a repository to private as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.update_repo_visibility(token=token, name=REPO_NAME, private=True)
+```
+
+## `Repository` 
+
+The `Repository` class allows you to push models or other repositories to the Hub. `Repository` is a wrapper over Git and Git-LFS methods, so make sure you have Git-LFS installed (see [here](https://git-lfs.github.com/) for installation instructions) and set up before you begin. The `Repository` class should feel familiar if you are already familiar with common Git commands. 
```

Comment:
I think I asked this before, but not sure. @LysandreJik do we already install git-lfs automatically when people pip install huggingface_hub?